### PR TITLE
Refactor of D3 Alignment tests. For initial review.

### PR DESCRIPTION
**Pull request by tchaffee**
#### Pre-Submission Checklist

- [x] Your pull request targets the `master` branch of freeCodeCamp/testable-projects-fcc
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] Your changes have been tested either locally or using a newly created CDN based on your fork's testable-projects-fcc/build/bundle.js file
- [ ] Working changes have been added to the build by running `npm run build`

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.

- [ ] Closes currently open issue: Closes #XXX

#### Description
- The first refactor (alignmentD3Tests.js) still left a lot to be desired because there were too many layers of functions and lots of unused parameters being passed from function to function. It was hard to understand. This time around we tried using a BDD approach by writing tests before writing code. It worked well, and the code is now more functional, and a lot smaller along with being easier to understand.
- Replaced the old alignmentD3Tests with alignment-D3 (contains only the core functions) and alignment-D3-support (contains functions specific to project tests).
- The main tests are written and can be found in test/alignment-D3.js. That's the best place to start to review this PR.
- You can run the tests just for this PR with `$(npm bin)/mocha -g 'D3 Alignment'`.
- A new issue will be created for some remaining tests that should be written in test/alignment-D3-support.js
- Added a .babelrc file to give mocha the babel config.
- Updated mocha.opts to use the babel transpiler.

### CODE REVIEW

File: src/assets/alignment-D3.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  afterTick = getNearestTick(
+    ticks,
+    (tickPosition, position) => tickPosition > position,
+    dimension,
+    position,
+    getTickPosition,
+    (prevPosition, position) => prevPosition < position
+  );
+
+  return [ beforeTick, afterTick ];
```
Comment (tbushman) [inline src/assets/alignment-D3.js]: The way accounting for axis sort order was handled before was to assign the lower value to beforeTick and vice versa. i.e. `beforeTick = Math.min(beforeTick, afterTick)`

Comment (tchaffee) [reply src/assets/alignment-D3.js]: @tbushman I'm pretty sure the way I calculate the beforeTick and afterTick already account for this. It doesn't matter what order the ticks are on the axis. For the beforeTick for example, it just looks for all the ticks with a value that lower than the shape value, and then it finds the tick closest to the shape value. I'll double check it when I get a chance.

**Comment by tchaffee** [comment]
@tbushman This is ready for final review and merge now.

**Comment by tbushman** [comment]
@tchaffee Sorry, but I'm still getting failing test number 8 on this pen (forked, with rawgit url in html `<script src>`): https://codepen.io/tbushman/pen/VMrOpj . This was passing previously.

**Comment by tchaffee** [comment]
@tbushman I'll look closer at your pen tomorrow, but if you have time in the meantime, can you try to figure out why it's failing against your CodePen but not the official CodePen? You can see in the checks (https://travis-ci.org/freeCodeCamp/testable-projects-fcc/builds/283306560?utm_source=github_status&utm_medium=notification) that all the tests passed... so I'm confused why the tests would pass but your forked CodePen isn't? I'm hoping we don't still have a problem with the tests being performed

**Comment by tbushman** [comment]
@tchaffee Right. The official fCC scatter-plot and heat-map Codepens do not have reverse-sorted y-axes.

**Comment by tchaffee** [comment]
@tbushman Well spotted. My test data for the reverse-sorted y-axis was wrong! It wasn't reverse sorted... So I've fixed the test data, the tests now fail and I'm now trying to figure out how to fix the code.

**Comment by tchaffee** [comment]
@tbushman, I've added some code to detect when an axis is in reverse order and to do the comparison differently when that is the case. This should be ready to merge now if everything works for you. If that's the case, any time you could spend giving the code a close look would be appreciated. I'm happy to keep improving this until everything is clear and well documented. Hopefully the automated tests go a long way toward documenting how things are supposed to work.

**Comment by tbushman** [comment]
@tchaffee thanks. Out here giving all of this stuff a close look. Great work here. Very creative functioning.  
  
Regarding the auto-tests, I'll discuss separately some use cases where I need input on how you'd test student projects.

**Comment by tbushman** [comment]
@tchaffee Could you please test this pen: https://codepen.io/tom-p-uk/full/ZeQEaE

**Comment by tchaffee** [comment]
@tbushman That fails with the months' axis, which is what I'm guessing you saw too? Thanks, I will look into this.

**Comment by tchaffee** [comment]
@tbushman This is actually a problem with the code in heat-maps-tests.js that tests if the `data-month` attribute is between 0 and 11. In this case the code is using `January`, `February`, etc. for the `data-month` attribute, and that's wrong. But our tests in heat-maps-tests.js do not catch it. I will create a separate issue for that bug.

**Comment by tchaffee** [comment]
I opened new issue #234 to fix this problem.

**Comment by tbushman** [comment]
@tchaffee thanks. LGTM!

**Pull request merged**